core/model/src/main/java/org/eclipse/sw360/antenna/model/util/ArtifactUtils.java

```diff
@@ -76,4 +78,23 @@ private ArtifactUtils() {
                         .map(fn -> new GenericArtifactCoordinates(fn, "-")),
                 artifact);
     }
+
+    public static ArtifactCoordinates createArtifactCoordinates(PackageURL packageURL) {
+        String type = packageURL.getType();
+
+        switch (type) {
+            case "maven":
+                return new MavenCoordinates(packageURL.getName(), packageURL.getNamespace(), packageURL.getVersion());
+            case "npm":
+                return new JavaScriptCoordinates(packageURL.getName(), packageURL.getNamespace(), packageURL.getVersion());
```

Comment:
I have added p2 as a type for BundleCoordinates although it is not a standardized type, but since we need it.